cmd/manager/render.go

```diff
@@ -115,22 +124,76 @@ func runRenderCmd(cmd *cobra.Command, args []string) {
-		log.Debugf("writing file: %s", assetRenderPath)
+		log.Infof("Writing file: %s", assetRenderPath)
 		err = ioutil.WriteFile(assetRenderPath, asset, 0644)
 		if err != nil {
 			log.WithError(err).Fatal("failed to write file")
 		}
 	}
 
-	// render static pod
-	if err := os.Mkdir(filepath.Join(ccoRenderDir, bootstrapManifestsDir), 0775); err != nil {
-		log.WithError(err).Fatal("error creating bootstrap-manifests directory")
+	if !operatorDisabled {
+		log.Info("Rendering static pod")
+		if err := os.Mkdir(filepath.Join(ccoRenderDir, bootstrapManifestsDir), 0775); err != nil {
+			log.WithError(err).Fatal("error creating bootstrap-manifests directory")
+		}
+		podPath := filepath.Join(ccoRenderDir, bootstrapManifestsDir, "cloud-credential-operator-pod.yaml")
+		podContent := fmt.Sprintf(podTemplate, renderOpts.ccoImage)
+		log.Infof("writing file: %s", podPath)
+		err = ioutil.WriteFile(podPath, []byte(podContent), 0644)
+		if err != nil {
+			log.WithError(err).Fatal("failed to write file")
+		}
+	} else {
+		log.Info("CCO disabled, skipping static pod manifest.")
+	}
+}
+
+// isDisabled will search through all the files in manifestsDir for a configmap indicating
+// whether or not CCO is disabled. In the absence of any configmap, it will return
+// the default disabled setting for the operator.
+func isDisabled() bool {
+
+	if renderOpts.manifestsDir == "" {
+		log.Debug("no manifests-dir param provided, using default operator settings")
+		return utils.OperatorDisabledDefault
 	}
-	podPath := filepath.Join(ccoRenderDir, bootstrapManifestsDir, "cloud-credential-operator-pod.yaml")
-	podContent := fmt.Sprintf(podTemplate, renderOpts.ccoImage)
-	log.Debugf("writing file: %s", podPath)
-	err = ioutil.WriteFile(podPath, []byte(podContent), 0644)
+
+	files, err := ioutil.ReadDir(renderOpts.manifestsDir)
 	if err != nil {
-		log.WithError(err).Fatal("failed to write file")
+		log.WithError(err).Errorf("failed to list files in %s, using defualt operator settings", renderOpts.manifestsDir)
+		return utils.OperatorDisabledDefault
 	}
+
+	for _, fInfo := range files {
+		// non-recursive checking of all files in the manifestDir
+		if fInfo.IsDir() {
+			continue
+		}
+
+		fullPath := filepath.Join(renderOpts.manifestsDir, fInfo.Name())
+		log.Debugf("checking file: %s", fullPath)
+		file, err := os.Open(fullPath)
+		if err != nil {
+			log.WithError(err).Warn("failed to open file while searching for configmap")
+			continue
+		}
+		decoder := yaml.NewYAMLOrJSONDecoder(file, 4096)
+		configMap := &corev1.ConfigMap{}
+		if err := decoder.Decode(configMap); err != nil {
+			log.WithError(err).Debug("failed to decode into configmap")
+			continue
+		}
+
+		if configMap.Namespace == minterv1.CloudCredOperatorNamespace && configMap.Name == minterv1.CloudCredOperatorConfigMap {
+
+			disabled, err := utils.CCODisabledCheck(configMap, &log.Logger{Level: log.GetLevel()})
```

Comment:
This does not set up the logger sufficiently and will cause a panic if the logger is ever used. Use `log.New` instead.